controllers/Backend.php

```diff
@@ -1,104 +1,352 @@
+           if(filter_has_var(INPUT_POST, 'content')){
+                $content = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'content')));
+            }
+
+            $posted = filter_input(INPUT_POST, 'public') ? "1" : "0";
+
+            $errors = [];
+
+            if(empty($title) || empty($content) || empty($chapo)){
+                $errors['empty'] = "Veuillez remplir tous les champs";
+            }
+            if(strlen($title) < 5){
+                $errors['title'] = "Votre titre doit contenir au moins 5 caractères.";
+            }
+            if(strlen($chapo) < 5){
+                $errors['chapo'] = "Votre chapô doit contenir au moins 5 caractères.";
+            }
+            if(strlen($title) < 5){
+                $errors['content'] = "Votre article doit contenir au moins 100 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(empty($errors)){
+                $manager_article->post($title,$chapo,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    $manager_article->post_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }else{
+                   header("Location:/liste-de-tous-les-articles");
+                }
+            }
+        }
+        else
+            $errors =[];
+   
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
+        
+        $myView = new View('write', 'backend');
+        $myView->render(array('errors' => $errors));
     }
 
     public function settings()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+        $modos = $manager_user->get_modos();
+ 
+        if(filter_has_var(INPUT_POST, 'submit')){
+                if(filter_has_var(INPUT_POST, 'name')){
+                    $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name')));
+                }
+                if(filter_has_var(INPUT_POST, 'email')){
+                    $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email')));
+                }
+                if(filter_has_var(INPUT_POST, 'email_again')){
+                    $email_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email_again', FILTER_VALIDATE_EMAIL)));
+                }
+                if(filter_has_var(INPUT_POST, 'role')){
+                    $role = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'role')));
+                }
+                
+                $token = UserManager::token(30);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class 'UserManager' in method 'settings'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)